go/store/types/serial_message.go

```diff
@@ -541,3 +519,51 @@ func (sm SerialMessage) writeTo(w nomsWriter, nbf *NomsBinFormat) error {
 func (sm SerialMessage) valueReadWriter() ValueReadWriter {
 	return nil
 }
+
+// init assigns the Doltgres-modifiable variables with their Dolt-specific functions. This must be done during init, as
+// assigning them directly to the variables causes an initialization cycle, and this is the only way to bypass it. This
+// is fine to do, as Doltgres is guaranteed to overwrite these functions due to Go's order of initialization.
+func init() {
+	RootValueHumanReadableStringAtIndentationLevel = rootValueHumanReadableStringAtIndentationLevel
+	RootValueWalkAddrs = rootValueWalkAddrs
+}
+
+// RootValueHumanReadableStringAtIndentationLevel returns the human readable string at the given indentation level for
+// root values. This is a variable as it's changed in Doltgres.
+var RootValueHumanReadableStringAtIndentationLevel func(sm SerialMessage, level int) string
+
+// RootValueWalkAddrs walks the given message using the given callback. This is a variable as it's changed in Doltgres.
+var RootValueWalkAddrs func(sm SerialMessage, nbf *NomsBinFormat, cb func(addr hash.Hash) error) error
+
+// rootValueHumanReadableStringAtIndentationLevel is Dolt's implementation of RootValueHumanReadableStringAtIndentationLevel.
+func rootValueHumanReadableStringAtIndentationLevel(sm SerialMessage, level int) string {
+	msg, _ := serial.TryGetRootAsRootValue(sm, serial.MessagePrefixSz)
+	ret := &strings.Builder{}
+	printWithIndendationLevel(level, ret, "{\n")
+	printWithIndendationLevel(level, ret, "\tFeatureVersion: %d\n", msg.FeatureVersion())
+	printWithIndendationLevel(level, ret, "\tForeignKeys: #%s\n", hash.New(msg.ForeignKeyAddrBytes()).String())
+	printWithIndendationLevel(level, ret, "\tTables: %s\n",
+		SerialMessage(msg.TablesBytes()).HumanReadableStringAtIndentationLevel(level+1))
+	printWithIndendationLevel(level, ret, "}")
+	return ret.String()
+}
+
+// rootValueWalkAddrs is Dolt's implementation of RootValueWalkAddrs.
+func rootValueWalkAddrs(sm SerialMessage, nbf *NomsBinFormat, cb func(addr hash.Hash) error) error {
```

Comment:
If this matters it should probably be on the interface. Seems like it's only used for debug printing though.